### PR TITLE
Mr 19 cov forest tab

### PR DESCRIPTION
**Pull request by BalancedmonkeY**
Work involved further modularising of the base app - so may want to view changes by individual commits, or most recent three, to see more detailed changes.

### CODE REVIEW

File: R/gemtc_analysis.R
Comment (tommorris168) [inline R/gemtc_analysis.R:1]: `paste0("Value for covariate ", model$model$regressor$variable, " set at ", cov_value)`
Can we have this displayed in the app as well as in the download?

**Review by tommorris168 — CHANGES_REQUESTED** [review]
All code looks good, and the things I can try all work. There are some things I can't try until they are implemented, e.g. sensitivity analysis.
Please change the function names to upper camel case where they aren't already (unless this interferes with existing code).

File: R/gemtc_analysis.R
Comment (tommorris168) [reply R/gemtc_analysis.R:1]: Just seen on Jira this is to be done later.

**Comment by BalancedmonkeY** [comment]
> All code looks good, and the things I can try all work. There are some things I can't try until they are implemented, e.g. sensitivity analysis. Please change the function names to upper camel case where they aren't already (unless this interferes with existing code).

I have changed some functions (and variable names) to follow the style guide. However, I haven't changed the Shiny module functions to upper camel case as they are a bit different. Janion hopes to chat about this on Friday.

File: R/bayesian/base tab/forest_plot_page.R
[the remark this answers is not in the document]
Comment (BalancedmonkeY) [reply R/bayesian/base tab/forest_plot_page.R:1]: This was historic code - probably from how it was built up. Changed to use outcome_measure and feed up through the modules.

[the remark this answers is not in the document]
Comment (BalancedmonkeY) [reply R/bayesian/base tab/forest_plot_page.R:1]: Done

[the remark this answers is not in the document]
Comment (BalancedmonkeY) [reply R/bayesian/base tab/forest_plot_page.R:1]: Done

File: R/bayesian/forest_plot_plus_stats.R
[the remark this answers is not in the document]
Comment (BalancedmonkeY) [reply R/bayesian/forest_plot_plus_stats.R:1]: Done

File: R/gemtc_analysis.R
Comment (BalancedmonkeY) [reply R/gemtc_analysis.R:1]: Yes, as it will be visible via the options above every tab - I didn't think it was necessary to be printed on the screen.

**Review by tommorris168 — CHANGES_REQUESTED** [review]
The step in `numericInput` isn't updating if you do a new analysis without closing the app, so please add the step change to the `updateNumericInput` functions in `bayesian_forest_plot_plus_stats_server` in _forest_plot_plus_stats.R_.

The new name of `gemtctau` in _fn_analysis.R_ will cause a conflict when merging into dev, because this function is no longer in _fn_analysis.R_. Might not be a problem.

I don't think the warnings in _covariate tab/forest_plot_page.R_ will ever be displayed, because the analysis is blocked when you select the radio button for one of the unimplemented analyses. If you want to ignore this for now I'm happy to approve, but the redundant code should be removed at some point.

**Comment by BalancedmonkeY** [comment]
> The step in `numericInput` isn't updating if you do a new analysis without closing the app, so please add the step change to the `updateNumericInput` functions in `bayesian_forest_plot_plus_stats_server` in _forest_plot_plus_stats.R_.
> 
> The new name of `gemtctau` in _fn_analysis.R_ will cause a conflict when merging into dev, because this function is no longer in _fn_analysis.R_. Might not be a problem.
> 
> I don't think the warnings in _covariate tab/forest_plot_page.R_ will ever be displayed, because the analysis is blocked when you select the radio button for one of the unimplemented analyses. If you want to ignore this for now I'm happy to approve, but the redundant code should be removed at some point.

Good spot on the `updateNumericInput` - thanks. I've also removed the warning for the covariate analysis as you're right, its not needed.
For now, the feature branch doesn't have those changes, so the renaming of `gemtctau` should be okay for now. But I have done some rearranging to help with the eventual merge.

**Comment by JanionNevill** [comment]
> The step in `numericInput` isn't updating if you do a new analysis without closing the app, so please add the step change to the `updateNumericInput` functions in `bayesian_forest_plot_plus_stats_server` in _forest_plot_plus_stats.R_.

You added steps to the min, but not the max. Was there a reason for this?

**Review by JanionNevill — CHANGES_REQUESTED** [review]
Just a minor question

**Comment by BalancedmonkeY** [comment]
> > The step in `numericInput` isn't updating if you do a new analysis without closing the app, so please add the step change to the `updateNumericInput` functions in `bayesian_forest_plot_plus_stats_server` in _forest_plot_plus_stats.R_.
> 
> You added steps to the min, but not the max. Was there a reason for this?
> 
> ### EDIT:
> After discussion, it was realised that the checks should be on the outcome measure, not the meta outcome. Please changes this to reflect better when the measure is logarithmic or linear.
> 
> While you're there....
> 
> * The formatting is inconsistent with spaces around the `=`.
> * The If statement should also have and `else` statement to throw an error if the outcome is not one of the supported options.
> * The reactive doesn't need to be assigned to a variable `x` at the beginning, it can just be called in each relevant location.

Implemented all this. However, for RD, MD, and SMD, the lower limit will step up by 0.1 first, then 1 onwards. I don't really know why it's doing this, and is something the app already does...

**Review by JanionNevill — APPROVED** [review]
Made minor changes. Otherwise all looks good.

**Review by JanionNevill — APPROVED** [review]
I've removed the default covariate value from the value panel. Everything else looks good.

**Review by tommorris168 — CHANGES_REQUESTED** [review]
For the box/button where you enter the covariate value, there needs to be some text to say that's what it is. Also, I'm not convinced it's in the best place. I would have put it lower down, say underneath the "choose type of regression coefficient" selection box. This might also sort out the alignment, because it's currently slightly too high. If other people prefer the current position I'm happy to leave it.

Can we reorder the options in the "choose type of regression coefficient" selection box? I would have put them in order of the number of parameters, i.e. shared, exchangeable, unrelated (or the reverse).

**Review by tommorris168 — APPROVED** [review]

**Pull request merged**